MediaBrowser.Providers/TV/TheTVDB/TvdbSeasonImageProvider.cs

```diff
@@ -62,91 +57,78 @@ public async Task<IEnumerable<RemoteImageInfo>> GetImages(BaseItem item, Cancell
             var season = (Season)item;
             var series = season.Series;
 
-            if (series != null && season.IndexNumber.HasValue && TvdbSeriesProvider.IsValidSeries(series.ProviderIds))
+            if (series == null || !season.IndexNumber.HasValue || !TvdbSeriesProvider.IsValidSeries(series.ProviderIds))
             {
-                var seriesProviderIds = series.ProviderIds;
-                var seasonNumber = season.IndexNumber.Value;
+                return new RemoteImageInfo[] { };
+            }
 
-                var seriesDataPath = await TvdbSeriesProvider.Current.EnsureSeriesInfo(seriesProviderIds, series.Name, series.ProductionYear, series.GetPreferredMetadataLanguage(), cancellationToken).ConfigureAwait(false);
+            var tvdbId = Convert.ToInt32(series.GetProviderId(MetadataProviders.Tvdb));
+            var seasonNumber = season.IndexNumber.Value;
+            var language = item.GetPreferredMetadataLanguage();
+            _tvDbClientManager.TvDbClient.AcceptedLanguage = language;
```

Comment:
_repeating so those issues won't get lost_

@cvium what about potential data race issues here? What if parallel requests would change accepted language to some different value?

There are also more similar looking places.

Reply:
Handled